lib/curl_ntlm_core.h

```diff
@@ -53,6 +53,10 @@
 #define USE_NTLM2SESSION 1
 #endif
 
+#if !defined(USE_NTLM2SESSION)
+#define USE_NTLM2SESSION 0
```

Comment:
Yes, exactly. All the macros defined in curl_ntlm_core.h and used in curl_ntlm_core.c and vauth/ntlm.c are checked with `#if` instead of `#ifdef`:
`USE_NTLM_V2`
`USE_NTRESPONSES`
`USE_NTLM2SESSION`
Seems like this is for historical reasons, they were always defined to 1 when introduced in 2006.

I'll glady change this in a separate pull request and remove the curl_ntlm_core.h changes from this one.